detector/vuls2/vendor.go

```diff
@@ -0,0 +1,240 @@
+		return latter
+	default:
+		return statusX
+	}
+}
+
+func mostPreferredTag(family string, segments []segmentTypes.Segment) segmentTypes.DetectionTag {
+	if len(segments) == 0 {
+		return segmentTypes.DetectionTag("")
+	}
+
+	s := slices.MaxFunc(segments, func(x, y segmentTypes.Segment) int {
+		return tagPreference(family, string(x.Tag)) - tagPreference(family, string(y.Tag))
+	})
+	return s.Tag
+}
+
+func resolvePackageByTag(family string, x, y pack) pack {
+	if tagPreference(family, string(x.tag)) < tagPreference(family, string(y.tag)) {
+		return y
+	}
+	return x
+}
+
+func resolveAdvisoryByTag(family string, x, y taggedAdvisory) taggedAdvisory {
+	if tagPreference(family, string(x.tag)) < tagPreference(family, string(y.tag)) {
+		return y
+	}
+	return x
+}
+
+func tagPreference(family string, tag string) int {
+	switch family {
+	case constant.RedHat, constant.CentOS:
+		switch {
+		case strings.HasSuffix(tag, "-including-unpatched"):
+			return 4
+		case strings.HasSuffix(tag, "-extras-including-unpatched"):
+			return 3
+		case strings.HasSuffix(tag, "-supplementary"):
+			return 2
+		default:
+			return 1
+		}
+	default:
+		return 1
+	}
+}
+
+func ignoresVulnerability(family string, v vulnerabilityTypes.Vulnerability, a *advisoryTypes.Advisory) bool {
+	switch family {
+	case constant.RedHat, constant.CentOS:
+		if a == nil && strings.Contains(v.Content.Description, "** REJECT **") {
+			return true
+		}
+		if a != nil && strings.Contains(a.Content.Description, "** REJECT **") {
+			return true
+		}
+		return false
+	default:
+		return false
+	}
+}
+
+func ignoresCriterion(family string, cn criterionTypes.FilteredCriterion) bool {
+	switch family {
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+		if cn.Criterion.Version.FixStatus != nil && cn.Criterion.Version.FixStatus.Class == fixstatusTypes.ClassUnfixed {
+			switch strings.ToLower(cn.Criterion.Version.FixStatus.Vendor) {
+			case "will not fix", "under investigation":
+				return true
+			}
+		}
+		return false
+	default:
+		return false
+	}
+}
+
+func ignoresWholeCriteria(family string, cn criterionTypes.FilteredCriterion) bool {
+	switch family {
+	case constant.RedHat, constant.CentOS, constant.Alma, constant.Rocky:
+		// Ignore whole criteria from root if kpatch-patch-* package is included.
+		if cn.Criterion.Type == criterionTypes.CriterionTypeVersion && cn.Criterion.Version != nil &&
+			cn.Criterion.Version.Package.Type == versioncriterionpackageTypes.PackageTypeBinary && cn.Criterion.Version.Package.Binary != nil &&
+			strings.HasPrefix(cn.Criterion.Version.Package.Binary.Name, "kpatch-patch-") {
+			return true
+		}
+		if cn.Criterion.Type == criterionTypes.CriterionTypeNoneExist && cn.Criterion.NoneExist != nil &&
+			cn.Criterion.NoneExist.Type == noneexistcriterionTypes.PackageTypeBinary && cn.Criterion.NoneExist.Binary != nil &&
+			strings.HasPrefix(cn.Criterion.NoneExist.Binary.Name, "kpatch-patch-") {
+			return true
+		}
```

Comment:
Would it be better to limit the data sources to which this processing applies?

Reply:
source ID filter is added https://github.com/future-architect/vuls/pull/2106/commits/532998318445cf1326bad0befb6ab98b3191cd00#diff-1be336d4b7528308cbd3a4c85b9bdb0938f40b2c973f26011f7048bd7352a7cdR139